collection-scripts/gather_haproxy_config

```diff
@@ -0,0 +1,22 @@
+#!/bin/bash
+
+BASE_COLLECTION_PATH="must-gather"
+HAPROXY_CONFIG_PATH=${OUT:-"${BASE_COLLECTION_PATH}/haproxy"}
+
+mkdir -p "${HAPROXY_CONFIG_PATH}"/
+
+function gather_haproxy_config {
+    echo "INFO: Gathering HAProxy config files"
+    INGRESS_CONTROLLERS=$(oc get ingresscontroller -n openshift-ingress-operator --no-headers -o custom-columns=":metadata.name")
+    for IC in ${INGRESS_CONTROLLERS}; do
+        PODS=$(oc get pods -n openshift-ingress --no-headers -o custom-columns=":metadata.name" --selector ingresscontroller.operator.openshift.io/deployment-ingresscontroller="${IC}")
+        for POD in ${PODS}; do
```

Comment:
Do we need to make this loop configurable, what happens if the ingress-controller pod is not an haproxy pod? Do we need to make this configurable or integrated with the pod (some data collection convention), so that this works with 3rd party ingress-controllers?

Reply:
3rd party ingress controllers wouldn't be able to make use of the OCP ingress controller API, and this scripts selects `PODS` by OCP ingress controller, so I think we are OK here.